provider/machinepool/machine_pool_resource.go

```diff
@@ -898,7 +898,6 @@ func (r *MachinePoolResource) populateState(object *cmv1.MachinePool, state *Mac
 			}
 		} else {
 			state.UseSpotInstances = types.BoolNull()
-			state.MaxReplicas = types.Int64Null()
```

Comment:
I believe this was incorrectly set at this section as it is also handled slightly below within the actual autoscaling resources